### ISSUE TITLE
Error in torch.set_default_tensor_type() documentation online (depreciated)

### ISSUE THREAD

**Issue by RichardSrn** [pytorch/pytorch#113646]
### 📚 The doc issue

Not sure it is the good place for reporting this, but the documentation at :
https://pytorch.org/docs/stable/generated/torch.set_default_tensor_type.html 
doesn't state that torch.set_default_tensor_type() has been **depreciated** as of torch 2.1.

### Suggest a potential alternative/fix

Redirect to torch.set_default_dtype() (https://pytorch.org/docs/stable/generated/torch.set_default_dtype.html)

**Comment by janeyx99** [comment]
Yes, it looks like https://github.com/pytorch/pytorch/pull/99521 did not change the docs. We would accept a PR fixing the docs for the function to redirect users to `set_default_dtype()`.

**Comment by RichardSrn** [comment]
Hi,
Well I'm glad I could contribute, and I am trying to make a PR (would be my first PR actually). So I have been reading the documentation on https://github.com/RichardSrn/pytorch/blob/main/CONTRIBUTING.md#writing-documentation , but I can't figure out how / where to add the modification... 

The only relevant file matching the string "set_default_tensor_type" seems to be ./pytorch/docs/source/torch.rst:21, but it is not where the actual documentation is.

If you have time to guide me I would be glad to make that PR, otherwise I will leave it to somebody else.

Thank you for developing PyTorch ! :)

**Comment by janeyx99** [comment]
A good tip is to search the codebase for `set_default_tensor_type` as documentation is often around where the code is implemented. Searching this repo for that yields https://github.com/pytorch/pytorch/blob/b88abb16748538a5cd81d2e23d506e4902266db1/torch/__init__.py#L663-L682
which seems like where you'd want to make the changes.

For the PR, not only will this doc need changing, all references to `set_default_tensor_type` should get redirected. (There are several others sprinkled across the codebase.)
